modules/xfem/src/base/XFEM.C

```diff
@@ -33,11 +33,11 @@ XFEM::XFEM (MooseApp & app) :
 XFEM::~XFEM ()
 {
   for (unsigned int i=0; i<_geometric_cuts.size(); ++i)
-  {
     delete _geometric_cuts[i];
-    _geometric_cuts[i] = NULL;
-  }
-  _geometric_cuts.clear();
+
+  std::map<unique_id_type, XFEMCutElem*>::iterator cemit = _cut_elem_map.begin();
```

Comment:
If you were worried about optimization (which isn't an issue for standard containers) you have it backwards. For some of the predicate-based iterators in libMesh you normally pre-declare the end iterator since that's the one that's compared against in every iteration. The begin iterator is only declared and set once per loop. This code also has the side effect of having that variable hanging around in the outer scope.